core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -1221,7 +1206,7 @@ private[kafka] class Processor(
   private def processDisconnected(): Unit = {
     selector.disconnected.keySet.forEach { connectionId =>
       try {
-        val remoteHost = ConnectionId.fromString(connectionId).getOrElse {
+        val remoteHost = ServerConnectionId.fromString(connectionId).orElseThrow { () =>
           throw new IllegalStateException(s"connectionId has unexpected format: $connectionId")
```

Comment:
Maybe `ServerConnectionId.fromString` can throw `IllegalStateException` instead of returning `Optional`?

Reply:
I was trying to maintain least refactor while keeping original context. I ll look into this as well.